ruby/ext/google/protobuf_c/extconf.rb

```diff
@@ -2,7 +2,9 @@
 
 require 'mkmf'
 
-$CFLAGS += " -O3 -std=c99 -Wno-unused-function -DNDEBUG "
+$CFLAGS += " -O3 -std=c99 -Wno-unused-function " +
+           "-Wno-declaration-after-statement -Wno-unused-variable " +
```

Comment:
What a pain! You would think that 16 years after C99 was ratified that it would be easier to integrate into modern build systems.

I may just bite the bullet at some point and convert upb to C89 + `stdint.h`. The one thing that worries me is that in C89 there is no way to statically initialize a non-first union member.